install/threesdk/cli.py

```diff
@@ -110,15 +110,66 @@ def version():
     Print version number
     """
     print(f"3sdk {_sdk.__version__}")
-    # from InstallTools import get_newest_version
-    dic = _sdk.InstallTools.Tools.get_newest_version()
     # campare with __version__
-    if not dic["up_to_date"]:
+    if not _sdk.InstallTools.Tools.is_latest_release():
+        dic = _sdk.InstallTools.Tools.get_latest_release()
         print(
-            f"Your SDK version is not up-to-date. Newest release is {dic['last_release']}\nPlease visit: {dic['last_release_url']}"
+            f"Your SDK version is not up-to-date. Newest release is {dic['latest_release']}\nPlease visit: {dic['latest_release_url']}\nOr run '3sdk --update'"
         )
 
 
+def update():
+    """
+    """
+    try:
+        from threesdk import container
+        from threesdk.InstallTools import JumpscaleInstaller, MyEnv, Tools
+    except ImportError:
+        from .core import container
+        from .core.InstallTools import JumpscaleInstaller, MyEnv, Tools
+
+    if not Tools.is_latest_release():
+        containers_names = container._containers.IT.DockerFactory.containers()
+        # pull all repos in containers
+        for item in containers_names:
+            name = item.name
+            print("Updating repositories in container ", name, "...")
+            container._containers.assert_container(name)
+            c = container._containers.get(name=name, explorer="none")
+            container_executor = c.executor
+            JumpscaleInstaller.repos_get(JumpscaleInstaller, pull=True, executor=container_executor)
+        import requests
+
+        # Update binary for host
+        print("Downloading 3sdk binary...")
+        latest_release = Tools.get_latest_release()
+        r = requests.get(latest_release["download_link"], allow_redirects=True)
+        if MyEnv.platform_is_linux or MyEnv.platform_is_osx:
+            Tools.file_write("/tmp/3sdk", r.content, True)
+            print("Done")
+            Tools.execute(f"chmod +x /tmp/3sdk")
+            _, bin_path, _ = Tools.execute(f"which 3sdk")
+            # to remove trailing \n
+            bin_path = bin_path[:-1]
+            # save backup
+            Tools.execute(f"cp -f {bin_path} /tmp/3sdk.bk")
+            # replace
+            try:
+                Tools.execute(f"mv -f /tmp/3sdk {bin_path}", interactive=True)
+            except:
+                Tools.execute(f"cp -f /tmp/3sdk.bk {bin_path}", interactive=True)
+                print(f"Failed to update binary, Can not replace binary in {bin_path}")
+        elif MyEnv.platform_is_windows:
+            update_path = f"{os.environ['USERPROFILE']}\\3sdk\\3sdk_update.exe"
+            print("Download done, installing now ..")
+            Tools.file_write(update_path, r.content, True)
```

Comment:
see comment of download in other section

Reply:
solved too :+1: